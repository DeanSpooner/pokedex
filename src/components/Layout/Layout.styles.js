import styled from "styled-components";

export const Title = styled.h1``;

export const ParentContainer = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
`;

export const MainContainer = styled.div`
  display: flex;
  flex-grow: 1;
`;

export const ChildContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 1%;
  padding: 20px;
  font-size: 20px;
  line-height: 20px;
  color: black;
  background-color: ${(props) => (props.isTitle ? "turquoise" : "#eab768")};
  border-radius: 10px;
  border: 10px solid ${(props) => (props.isTitle ? "aliceblue" : "#f5cc8b")};
  flex-grow: ${(props) => (props.isMain ? 1 : 0)};
`;
