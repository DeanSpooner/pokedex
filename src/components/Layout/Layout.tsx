import * as React from "react";
import Link from "next/link";
import {
  ChildContainer,
  MainContainer,
  ParentContainer,
  Title,
} from "./Layout.styles.js";

const Layout = () => {
  return (
    <ParentContainer>
      <ChildContainer isTitle>
        <Title>Dean's Pok√©dex</Title>
      </ChildContainer>
      <MainContainer>
        <ChildContainer>
          <Link href="/pokemon">
            <a>Kanto</a>
          </Link>
        </ChildContainer>
        <ChildContainer>
          <Link href="/newerpokemon">
            <a>Johto</a>
          </Link>
        </ChildContainer>
      </MainContainer>
      <ChildContainer isFooter>By Dean Spooner</ChildContainer>
    </ParentContainer>
  );
};

export default Layout;
