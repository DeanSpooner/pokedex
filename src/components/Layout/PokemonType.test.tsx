/** @jest-environment jsdom */

import React from "react";
import { render } from "@testing-library/react";
import Layout from "./Layout";

describe("Layout", () => {
  it("matches snapshot", () => {
    const { getByTestId } = render(<Layout />);
    expect(getByTestId("Layout")).toMatchSnapshot();
  });
});
