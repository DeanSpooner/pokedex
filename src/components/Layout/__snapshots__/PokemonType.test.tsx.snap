// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<div
  class="Layoutstyles__ParentContainer-sc-3a4sff-1 gJOuOp"
  data-testid="Layout"
>
  <div
    class="Layoutstyles__ChildContainer-sc-3a4sff-3 kHwKUm"
  >
    <h1
      class="Layoutstyles__Title-sc-3a4sff-0"
    >
      Dean's Pok√©dex
    </h1>
  </div>
  <div
    class="Layoutstyles__MainContainer-sc-3a4sff-2 fnmlQn"
  >
    <div
      class="Layoutstyles__ChildContainer-sc-3a4sff-3 jdoyEl"
    >
      <a
        href="/pokemon"
      >
        Kanto
      </a>
    </div>
    <div
      class="Layoutstyles__ChildContainer-sc-3a4sff-3 jdoyEl"
    >
      <a
        href="/newerpokemon"
      >
        Johto
      </a>
    </div>
  </div>
  <div
    class="Layoutstyles__ChildContainer-sc-3a4sff-3 jdnCdp"
  >
    By Dean Spooner
  </div>
</div>
`;
